scuttlebutt-handshake/src/main/java/net/consensys/cava/scuttlebutt/handshake/vertx/SecureScuttlebuttVertxServer.java

```diff
@@ -92,6 +118,15 @@ private void handleMessage(Buffer buffer) {
     }
   }
 
+  private int getBodyLength(Bytes rpcHeader) {
+    Bytes size = rpcHeader.slice(1, 4);
+    int bodySize = ((size.get(0) & 0xFF) << 8)
+        + ((size.get(1) & 0xFF) << 8)
+        + ((size.get(2) & 0xFF) << 8)
+        + ((size.get(3) & 0xFF));
```

Comment:
you can use size.toInt() if you would like to get the int.

I'm not quite sure you're shifting the bits enough. Should be 24, 16, 8, 0 shifts.